charts/karpenter/templates/role.yaml

```diff
@@ -10,25 +10,34 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 rules:
-  - apiGroups: [""]
-    resources: ["configmaps"]
-    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
-  - apiGroups: [""]
-    resources: ["configmaps/status"]
-    verbs: ["get", "update", "patch"]
-  - apiGroups: [""]
-    resources: ["events"]
-    verbs: ["create"]
+  # Read
   - apiGroups: ["coordination.k8s.io"]
     resources: ["leases"]
-    verbs: ["create", "get", "patch", "update", "watch"]
+    verbs: ["get", "watch"]
   - apiGroups: [""]
-    resources: ["namespaces"]
+    resources: ["configmaps", "namespaces", "secrets"]
     verbs: ["get", "list", "watch"]
```

Comment:
Can we scope this down to the specific ConfigMaps (logging, batchingConfig) and Secrets that we need via `resourceNames` ?

Reply:
I believe you can't restrict list requests by resource name, but we need list/watch for the informer. Since these are read permissions, I think this is acceptable.

Reply:
Created the role 
```
- apiGroups: [""]
    resources: ["configmaps"]
    verbs: ["get", "list", "watch"]
    resourceNames:
      - karpenter-global-settings
      - karpenter-leader-election
      - config-logging
```
```
karpenter-6746fb6c66-m7ttl controller E0621 17:29:47.216205       1 reflector.go:138] k8s.io/client-go@v0.21.4/tools/cache/reflector.go:167: Failed to watch *v1.ConfigMap: failed to list *v1.ConfigMap: configmaps is forbidden: User "system:serviceaccount:karpenter:karpenter" cannot list resource "configmaps" in API group "" in the namespace "karpenter"
```

Get is subsumed by list, so I don't see the value in restricting get.